arangod/Auth/UserManager.h

```diff
@@ -23,175 +23,117 @@
 
   // Will trigger a global reload and block until the versions are
   // incremented at least once. To ensure that the internal thread executed
   // loadFromDB at least once.
-  void triggerCacheRevalidation();
+  virtual void triggerCacheRevalidation() = 0;
 
   // Create the root user with a default password, will fail if the user
   // already exists. Only ever call if you can guarantee to be in charge
-  void createRootUser();
+  virtual void createRootUser() = 0;
 
-  velocypack::Builder allUsers();
+  virtual velocypack::Builder allUsers() = 0;
   // Add user from arangodb
-  Result storeUser(bool replace, std::string const& user,
-                   std::string const& pass, bool active,
-                   velocypack::Slice extras);
+  virtual Result storeUser(bool replace, std::string const& user,
+                           std::string const& pass, bool active,
+                           velocypack::Slice extras) = 0;
 
   // Enumerate list of all users
-  Result enumerateUsers(std::function<bool(auth::User&)>&&,
-                        bool retryOnConflict);
+  virtual Result enumerateUsers(std::function<bool(User&)>&&,
+                                bool retryOnConflict) = 0;
 
   // Update specific user
-  Result updateUser(std::string const& user, UserCallback&&);
+  virtual Result updateUser(std::string const& user, UserCallback&&) = 0;
 
   // Access user without modifying it
-  Result accessUser(std::string const& user, ConstUserCallback&&);
+  virtual Result accessUser(std::string const& user, ConstUserCallback&&) = 0;
 
   // does this user exists in the db
-  bool userExists(std::string const& user);
+  virtual bool userExists(std::string const& user) = 0;
 
   // Serialize user into legacy format for REST API
-  velocypack::Builder serializeUser(std::string const& user);
+  virtual velocypack::Builder serializeUser(std::string const& user) = 0;
 
   // Remove one user or all users
-  Result removeUser(std::string const& user);
-  Result removeAllUsers();
+  virtual Result removeUser(std::string const& user) = 0;
+  virtual Result removeAllUsers() = 0;
 
   // Convenience methods to check a password or access token
-  bool checkCredentials(std::string const& username, std::string const& token,
-                        std::string& un);
+  virtual bool checkCredentials(std::string const& username,
+                                std::string const& token, std::string& un) = 0;
 
-  auth::Level databaseAuthLevel(std::string const& username,
-                                std::string const& dbname,
-                                bool configured = false);
-  auth::Level collectionAuthLevel(std::string const& username,
+  virtual Level databaseAuthLevel(std::string const& username,
                                   std::string const& dbname,
-                                  std::string_view coll,
-                                  bool configured = false);
+                                  bool configured) = 0;
```

Comment:
No need to change anything; but a you could add a (non-virtual) function `Level databaseAuthLevel(std::string const& username, std::string const& dbname);`, that calls the original one with false as a third parameter, to emulate the default.